go.mod

```diff
@@ -46,22 +47,27 @@ require (
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/shopspring/decimal v1.4.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
-	golang.org/x/crypto v0.23.0 // indirect
+	github.com/x448/float16 v0.8.4 // indirect
+	golang.org/x/crypto v0.24.0 // indirect
 	golang.org/x/exp v0.0.0-20240525044651-4c93da0ed11d // indirect
-	golang.org/x/net v0.25.0 // indirect
-	golang.org/x/oauth2 v0.20.0 // indirect
-	golang.org/x/sys v0.20.0 // indirect
-	golang.org/x/term v0.20.0 // indirect
-	golang.org/x/text v0.15.0 // indirect
+	golang.org/x/net v0.26.0 // indirect
+	golang.org/x/oauth2 v0.21.0 // indirect
+	golang.org/x/sys v0.21.0 // indirect
+	golang.org/x/term v0.21.0 // indirect
+	golang.org/x/text v0.16.0 // indirect
 	golang.org/x/time v0.5.0 // indirect
-	google.golang.org/protobuf v1.34.1 // indirect
+	google.golang.org/protobuf v1.34.2 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	k8s.io/apiextensions-apiserver v0.29.5 // indirect
-	k8s.io/klog/v2 v2.120.1 // indirect
+	k8s.io/apiextensions-apiserver v0.31.0 // indirect
+	k8s.io/klog/v2 v2.130.1 // indirect
 	k8s.io/kube-openapi v0.0.0-20240521193020-835d969ad83a // indirect
-	k8s.io/utils v0.0.0-20240502163921-fe8a2dddb1d0 // indirect
+	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+// This can be removed when upgrading to v0.32.0. This replace is to include this fix:
+// https://github.com/kubernetes/client-go/commit/2176e80333ddd9966b37758f8417bf72ceb61d5f
+replace k8s.io/client-go => k8s.io/client-go v0.0.0-20240815135022-d63a65fbe7d4
```

Comment:
Rather than `replace`, can you just get the commit and see what happens?
```
go get k8s.io/client-go@2176e80333ddd9966b37758f8417bf72ceb61d5f
```

Reply:
(I updated the SHA--not sure why the SHA is different in the GitHub link and the Go pseudo-version. I guess something about the K8s release process?

Reply:
@dhaiducek it didn't work on kubernetes-dependency-watches because the `master` branch for client-go doesn't share a parent commit with a tagged version so it just comes up as a `0.0.0` version and that messes up other dependencies that need client-go.

I think it'll be the same here. We actually don't need the replace here but could just put the replace on the propagator, but I thought it'd be clearer if it was done everywhere.

Reply:
Oh, I see. That mirroring thing Kubernetes has going on makes things odd.

Reply:
I've come to discover that replaces are somewhat annoying because they're more difficult to maintain and break `go install`. So with this, there won't be a way to `go install` the `template-resolver` outside of the cloning the repo. Good thing we have the `policy-cli` on its way. 🙂

Reply:
Okay, let me remove that replace then. I'll just replace it in the Propagator.